src/lpc.jl

```diff
@@ -59,7 +66,7 @@ function implements the mathematics described in [1].
 """
 function lpc{T <: Number}(x::AbstractVector{T}, p::Int, ::LPCLevinson)
     R_xx = xcorr(x,x)[length(x):end]
-    a = zeros(p,p)
+    a = zeros(T,p,p)
```

Comment:
Good catch

Reply:
Is this accurate for integer `T`?

Reply:
Ah I wasn't even thinking about fixed point arithmetic. That would be cool.

My point with these comments is that the current PR will error if `x` is an abstract vector with `Integer` elements, because of the type assertions and container types used in the code. I think you should either change the type assertions and container types to reflect the actual types, or restrict the type bounds in the method definitions to types that will work as you expect (e.g. `AbstractFloat`).

Because restricting the type bounds will change the API for this function, I think it would be best to be more careful about the container types. If `T = eltype(x)` and `T <: AbstractFloat` or `T <: FixedPoint` then `prediciton_error`, `reflection_coeffs`, and `a` will be type `T` (or have eltype `T`). If `T <: Integer` then `prediction_error` and `reflection_coeffs` will be type `Float64`, and the Levinson algorithm will error because `xcorr` errors with integer inputs.

Reply:
Fixed-Point is quite common for LPC implementations, it's an import use-case as LPC is used in some analysis and compression algorithms in some embedded systems with Integer-Only hardware (still I believe); the few integer LPC's I've seen had the Fixed-Point arithmetics inlined by hand (for example in Speex), but this could actually be an interesting use case to see whether we can produce efficient machine code with an abstract type here (might change the abstraction for that). Anyhow, I think I've even tested this with Rational before, it's quite cool that it works.